airbyte-ci/connectors/pipelines/pipelines/dagger/containers/java.py

```diff
@@ -158,22 +161,31 @@ async def with_airbyte_java_connector(context: ConnectorContext, connector_java_
             )
         )
     )
-
-    if (
+    # TODO: remove the condition below once all connectors have a base image declared in their metadata.
+    if "connectorBuildOptions" in context.connector.metadata and "baseImage" in context.connector.metadata["connectorBuildOptions"]:
+        base_image_address = context.connector.metadata["connectorBuildOptions"]["baseImage"]
+        context.logger.info(f"Using base image {base_image_address} from connector metadata to build connector.")
+        base = context.dagger_client.container(platform=build_platform).from_(base_image_address)
+    elif (
         context.connector.supports_normalization
         and DESTINATION_NORMALIZATION_BUILD_CONFIGURATION[context.connector.technical_name]["supports_in_connector_normalization"]
     ):
-        base = with_integration_base_java_and_normalization(context, build_platform)
-        entrypoint = ["/airbyte/run_with_normalization.sh"]
+        context.logger.warn(
+            f"Connector {context.connector.technical_name} has in-connector normalization enabled. This is supposed to be deprecated. "
+            f"Please declare a base image address in the connector metadata.yaml file (connectorBuildOptions.baseImage)."
+        )
+        base = with_integration_base_java_and_normalization(context, build_platform).with_entrypoint(["/airbyte/run_with_normalization.sh"])
     else:
-        base = with_integration_base_java(context, build_platform)
-        entrypoint = ["/airbyte/base.sh"]
+        context.logger.warn(
+            f"Connector {context.connector.technical_name} does not declare a base image in its connector metadata. "
+            f"Please declare a base image address in the connector metadata.yaml file (connectorBuildOptions.baseImage)."
+        )
+        base = with_integration_base_java(context, build_platform).with_entrypoint(["/airbyte/base.sh"])
 
     connector_container = (
         base.with_workdir("/airbyte")
         .with_env_variable("APPLICATION", application)
         .with_mounted_directory("built_artifacts", build_stage.directory("/airbyte"))
         .with_exec(sh_dash_c(["mv built_artifacts/* ."]))
-        .with_entrypoint(entrypoint)
```

Comment:
just checking: this is safe to remove? I don't actually know

Reply:
Yes it it, I set it in the base image. And if the base image code path is not taken I added the entrypoint explicitely on line 177 and 183